server/src/test/java/io/crate/execution/engine/pipeline/ProjectionToProjectorVisitorTest.java

```diff
@@ -136,25 +136,42 @@ public void testSimpleTopNProjection() throws Exception {
     }
 
     @Test
-    public void testSortingTopNProjection() throws Exception {
+    public void testSortingLimitAndOffsetProjection() throws Exception {
         List<Symbol> outputs = Arrays.asList(Literal.of("foo"), new InputColumn(0), new InputColumn(1));
-        OrderedTopNProjection projection = new OrderedTopNProjection(10, 0, outputs,
-            Arrays.asList(new InputColumn(0), new InputColumn(1)),
-            new boolean[]{false, false},
-            new boolean[]{false, false}
+        OrderedLimitAndOffsetProjection projection = new OrderedLimitAndOffsetProjection(10, 0, outputs,
+                                                                                         Arrays.asList(new InputColumn(0), new InputColumn(1)),
+                                                                                         new boolean[]{false, false},
+                                                                                         new boolean[]{false, false}
         );
         Projector projector = visitor.create(
             projection, txnCtx, RamAccounting.NO_ACCOUNTING, memoryManager, UUID.randomUUID());
-        assertThat(projector, instanceOf(SortingTopNProjector.class));
+        assertThat(projector, instanceOf(SortingLimitAndOffsetProjector.class));
     }
 
     @Test
-    public void testTopNProjectionToSortingProjector() throws Exception {
+    public void testNoLimitNorOffsetProjection() throws Exception {
+        LimitAndOffsetProjection projection = new LimitAndOffsetProjection(-1, 10, Collections.singletonList(DataTypes.LONG));
+
+        Projector projector = visitor.create(
+            projection, txnCtx, RamAccounting.NO_ACCOUNTING, memoryManager, UUID.randomUUID());
+        assertThat(projector, instanceOf(SimpleOffsetProjector.class));
+
+        TestingRowConsumer consumer = new TestingRowConsumer();
```

Comment:
Maybe we should remove this test, and the similar ones from the class, as they are tested in the respective Projectors?

Reply:
If it's already tested elsewhere there is no need to test the same thing here too

Reply:
Thx, I will double check it and maybe remove in a separate PR (I mean the existing tests)